pkg/amd/manifest/firmware.go

```diff
@@ -91,15 +100,18 @@ func parsePSPFirmware(firmware Firmware) (*PSPFirmware, error) {
 			if entry.LocationOrValue != 0 && entry.LocationOrValue < uint64(len(image)) {
 				pspDirectoryLevel2, length, err := ParsePSPDirectoryTable(image[entry.LocationOrValue:])
 				if err == nil {
-					result.PSPDirectoryLevel2 = pspDirectoryLevel2
-					result.PSPDirectoryLevel2Range.Offset = entry.LocationOrValue
-					result.PSPDirectoryLevel2Range.Length = length
+					result.PSPDirectories[0].PSPDirectoryLevel2 = pspDirectoryLevel2
+					result.PSPDirectories[0].PSPDirectoryLevel2Range.Offset = entry.LocationOrValue
+					result.PSPDirectories[0].PSPDirectoryLevel2Range.Length = length
 				}
 			}
 			break
 		}
 	}
 
+	result.BIOSDirectories = []BIOSDir{}
```

Comment:
not sure I understand this. you can append to a nil slice. But 112 or 113 seem redundant. But you wanted two elements or ...?

Reply:
I don't know enough Go, will see about your suggestion.